flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/FileStoreCommitImpl.java

```diff
@@ -477,12 +477,20 @@ private boolean tryCommitOnce(
         List<ManifestFileMeta> newMetas = new ArrayList<>();
         List<ManifestFileMeta> changelogMetas = new ArrayList<>();
         try {
+            long previousChangesRecordCount = 0L;
             if (latestSnapshot != null) {
+                List<ManifestFileMeta> previousManifests =
+                        latestSnapshot.dataManifests(manifestList);
                 // read all previous manifest files
-                oldMetas.addAll(latestSnapshot.readAllDataManifests(manifestList));
+                oldMetas.addAll(previousManifests);
+                previousChangesRecordCount =
+                        latestSnapshot.version() >= Snapshot.CURRENT_VERSION
```

Comment:
Create a util method: `totalRecordCount(Snapshot snapshot)`.
And I think maybe just judge that field is null instead of snapshot version?